src/main/java/org/jabref/logic/importer/fetcher/INSPIREFetcher.java

```diff
@@ -78,4 +82,43 @@ public void doPostCleanup(BibEntry entry) {
     public Parser getParser() {
         return new BibtexParser(importFormatPreferences, new DummyFileUpdateMonitor());
     }
+
+    @Override
+    public List<BibEntry> performSearch(BibEntry entry) {
+        List<BibEntry> results = new ArrayList<>();
+        String doi = entry.getField(StandardField.DOI).orElse(null);
+
+        if (doi == null) {
+            return results;
+        }
+
+        String url = INSPIRE_EXTERNAL_HOST + doi;
+
+        try {
+
+            URL obj = new URL(url);
+            URLDownload download = getUrlDownload(obj);
```

Comment:
You can simplify this even more: Just download.asString() gives you the resoure as string.